Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionMapsView.cs

```diff
@@ -57,43 +62,49 @@ public ActionMapsView(VisualElement root, StateContainer stateContainer)
             m_ListView.RegisterCallback<ValidateCommandEvent>(OnValidateCommand);
 
             CreateSelector(s => new ViewStateCollection<string>(Selectors.GetActionMapNames(s)),
-                (actionMapNames, state) => new ViewState(Selectors.GetSelectedActionMap(state), actionMapNames));
+                (actionMapNames, state) => new ViewState(
+                    Selectors.GetSelectedActionMap(state), actionMapNames?.ToList() ?? new List<string>()));
 
             m_AddActionMapButton = root.Q<Button>("add-new-action-map-button");
             m_AddActionMapButton.clicked += AddActionMap;
             ContextMenu.GetContextMenuForActionMapListView(this, m_ListView.parent);
         }
 
-        public override void RedrawUI(ViewState viewState)
+        protected override void RedrawUI(ViewState viewState)
         {
-            m_ListView.itemsSource = viewState.actionMapNames?.ToList() ?? new List<string>();
-            if (viewState.selectedActionMap.HasValue)
+            m_ListView.itemsSource = viewState.actionMapNames;
+            m_ListView.Rebuild();
+
+            // Update view to reflect model selection
+            var desiredSelectedIndex = viewState.selectedActionMapIndex;
+            if (desiredSelectedIndex >= 0 && desiredSelectedIndex != m_ListView.selectedIndex)
             {
-                var indexOf = viewState.actionMapNames.IndexOf(viewState.selectedActionMap.Value.name);
-                m_ListView.SetSelection(indexOf);
+                m_ListView.SetSelection(desiredSelectedIndex);
+                m_ListView.ScrollToItem(desiredSelectedIndex);
             }
-            m_ListView.Rebuild();
-            RenameNewActionMaps();
         }
 
         public override void DestroyView()
         {
             m_AddActionMapButton.clicked -= AddActionMap;
         }
 
-        private void RenameNewActionMaps()
+        private void RenameNewActionMap()
         {
-            if (!m_EnterRenamingMode)
-                return;
-            m_ListView.ScrollToItem(m_ListView.selectedIndex);
             var element = m_ListView.GetRootElementForIndex(m_ListView.selectedIndex);
+            m_ListView.ScrollToItem(m_ListView.selectedIndex);
             ((InputActionMapsTreeViewItem)element).FocusOnRenameTextField();
-            m_EnterRenamingMode = false;
         }
 
         private void DeleteActionMap(int index)
         {
-            Dispatch(Commands.DeleteActionMap(index));
+            Dispatch(Commands.DeleteActionMap(index), () =>
+            {
+                // WORKAROUND: For some reason after delete (haven' identified a pattern), m_ListView would
+                // lose focus preventing further navigation or commands associated with the view.
+                // Hence we explicitly reclaim focus here post the delete command as a workaround.
+                m_ListView.Focus();
```

Comment:
FYI: This is [similar to what I did for deleting ActionTree item](https://github.com/Unity-Technologies/InputSystem/pull/1812/files#r1444433459).

Reply:
I haven't asked them but probably should. Did you follow up on this @timkeo ? 
However, I think its not uncommon behavior on delete operations in UI frameworks to move focus to the parent control/container. E.g. select a file in Explorer, press delete. Tentative focus in moved to the directory window (container) and hence pressing down key selects the first child element in the list. However, here we try to do something different due to the UI design of this window which would otherwise make Actions panel and Bindings panel empty after delete due to missing selection. I believe this is more similar to how Finder in macOS behaves.